packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -40,17 +40,35 @@ function adoptChildNode(node: Node, fn: ShadowRootResolver, shadowToken: string
     }
 }
 
+function disownChildNode(node: Node) {
+    // Note: this will remove the node owner key.
+    setShadowRootResolver(node, function() {} as ShadowRootResolver);
+
+    if (node instanceof Element) {
+        setShadowToken(node, undefined);
+        const childNodes = getInternalChildNodes(node);
+        for (let i = 0, len = childNodes.length; i < len; i += 1) {
+            const child = childNodes[i];
+            disownChildNode(child);
+        }
+    }
```

Comment:
Seems like we could reuse `adoptChildNode` for this and rely on tests to make sure the behavior is what we want?
```
function disownChildNode(node: Node) {
    adoptChildNode(node, function() {}, undefined);
}
```

Reply:
I agree with @ekashida here, you should reuse `adoptChildNode` because you have to recursively assign the keys to the children nodes, but there is still a problem:

* you still have a function that represents the resolver, that function is used in some places instead of the key, and in fact, I want to remove the keys everywhere, and focus on the resolver function instead, which is faster, and more efficient. I think it is still going to work fine since the function can be defined in a way that the metadata returned by that function is consistent with the document as the owner. But let's make sure that we understand this very well, I don't want to jeopardy the possibility to remove the keys entirely.

Reply:
thanks! yeah, now i defined a function that will hold the metadata of "non shadowed" nodes; if they are re-attached to another shadow, the resolver will be overwritten with the one from the shadow root, and if they are re-attached to another element that is not shadowed; this metadata will still be valid, like you mention.